Philips.CodeAnalysis.SecurityAnalyzers/LicenseAnalyzer.cs

```diff
@@ -1,43 +1,742 @@
+			{
+				// Report diagnostic about the error instead of silently failing
+				ReportDebugDiagnostic(context, $"Failed to analyze package licenses: {ex.Message}");
+			}
+		}
+
+		private static bool IsTestProject(Compilation compilation)
+		{
+			// Check if this is a test project by looking for common test framework references
+			var referencedAssemblyNames = new HashSet<string>(
+				compilation.ReferencedAssemblyNames.Select(name => name.Name),
+				StringComparer.OrdinalIgnoreCase);
+
+			return referencedAssemblyNames.Contains("Microsoft.VisualStudio.TestPlatform.TestFramework") ||
+				   referencedAssemblyNames.Contains("MSTest.TestFramework") ||
+				   referencedAssemblyNames.Contains("NUnit.Framework") ||
+				   referencedAssemblyNames.Contains("xunit") ||
+				   referencedAssemblyNames.Contains("xunit.core");
+		}
+
+		private void AnalyzePackagesFromAssetsFile(CompilationAnalysisContext context, HashSet<string> allowedLicenses)
+		{
+			// Get project.assets.json path from analyzer config options
+			var assetsFilePath = TryFindAssetsFileFromSourcePaths(context);
+			if (string.IsNullOrEmpty(assetsFilePath) || !File.Exists(assetsFilePath))
+			{
+				// Report diagnostic that project.assets.json was not found
+				ReportDebugDiagnostic(context, "Could not find project.assets.json file for license analysis");
+				return;
+			}
+
+			// Report where we found the file (but not the detailed search diagnostics when successful)
+			ReportDebugDiagnostic(context, $"Found {ProjectAssetsFileName} at: {assetsFilePath}");
+
+			// Load and parse project.assets.json manually to avoid external dependencies
+			List<PackageInfo> packages;
+			try
+			{
+				// Parse project.assets.json manually for better analyzer compatibility
+				packages = ParseProjectAssetsFile(assetsFilePath);
+			}
+			catch (Exception ex)
+			{
+				// Report diagnostic about parsing failure instead of silently failing
+				ReportDebugDiagnostic(context, $"Failed to parse project.assets.json: {ex.Message}");
+				return;
+			}
+
+			if (packages == null || packages.Count == 0)
+			{
+				return;
+			}
+
+			// Load or create license cache
+			Dictionary<string, PackageLicenseInfo> licenseCache = LoadLicenseCache(Path.GetDirectoryName(assetsFilePath));
+
+			// Analyze each package
+			foreach (PackageInfo package in packages)
+			{
+				if (package.Type != "package")
+				{
+					continue;
+				}
+
+				// Get license information for this package
+				var licenseInfo = GetPackageLicenseInfo(package, licenseCache);
+
+				// Report the license that was found for each reference (for diagnostic purposes)
+				var displayLicense = string.IsNullOrEmpty(licenseInfo) ? "unknown" : licenseInfo;
+				ReportDebugDiagnostic(context, $"Package {package.Name} {package.Version ?? "unknown"}: License = {displayLicense}");
+
+				// Check if license is acceptable
+				if (!string.IsNullOrEmpty(licenseInfo) && !IsLicenseAcceptable(licenseInfo, allowedLicenses))
```

Comment:
In testing on multiple projects, PH2155_INFO is being reported as informational, but not PH2155 diagnostics are being created (even though they seemingly should right now). I confirmed in Solution Explorer under dependencies that PH2155 "Avoid Packages with unacceptable licenses" is loaded and configured as error. We can temporarily remove this if clause to make sure I'm getting something reported alongside PH2155_INFO.